lib/bundler/plugin_dsl.rb

```diff
@@ -0,0 +1,49 @@
+require 'bundler/dsl'
+module Bundler
+  class PluginDsl
+
+    def self.evaluate(gemfile, lockfile, unlock)
+      builder = new
+      builder.eval_gemfile_for_plugins(gemfile)
+      builder.to_plugin_definition(lockfile, unlock)
+    end
+
+    def initialize
+      @dsl = Dsl.new
+      methods = Dsl.instance_methods(false)
+      methods.delete(:plugin)
+
+      self.class.instance_eval do
```

Comment:
If we subclass the existing Dsl, it will run as the definition. This "pre flight" Dsl is used to only process 

`plugin "whatever"`

and ignore the rest of the definition. Hence, we need all the methods defined as the original Dsl but they shouldn't do anything. Hence the empty methods.